File: src/Support/Clock.php
<?php

declare(strict_types=1);

namespace Meridian\Support;

/**
 * Process-wide clock. Tests may pin the time with Clock::setFixed().
 */
final class Clock
{
    private static ?\DateTimeImmutable $fixed = null;

    public static function now(): \DateTimeImmutable
    {
        return self::$fixed ?? new \DateTimeImmutable('now');
    }

    public static function setFixed(?\DateTimeImmutable $now): void
    {
        self::$fixed = $now;
    }
}

File: src/main/java/cep/time/Clock.java
package cep.time;

/** 可注入的时钟：生产环境用系统时间，测试用固定/虚拟时间。 */
public interface Clock {
    long nowMillis();

    static Clock system() {
        return System::currentTimeMillis;
    }

    static Clock fixed(long t) {
        return () -> t;
    }
}
